tests/components/influxdb/test_init.py

```diff
@@ -409,340 +394,326 @@ async def test_event_listener_states(
-            BASE_V2_CONFIG,
-            _get_write_api_mock_v2,
-            influxdb.API_VERSION_2,
-        ),
-    ],
-    indirect=["mock_client", "get_mock_call"],
-)
-async def test_event_listener_blacklist(
-    hass, mock_client, config_ext, get_write_api, get_mock_call
-):
-    """Test the event listener against a blacklist."""
-    handler_method = await _setup(hass, mock_client, config_ext, get_write_api)
 
-    for entity_id in ("ok", "blacklisted"):
+def execute_filter_test(hass, tests, listener, write_api, get_mock_call):
+    """Execute all tests for a given filtering test."""
+    for test in tests:
+        domain, entity_id = split_entity_id(test.id)
         state = MagicMock(
             state=1,
-            domain="fake",
-            entity_id=f"fake.{entity_id}",
+            domain=domain,
+            entity_id=test.id,
             object_id=entity_id,
             attributes={},
         )
         event = MagicMock(data={"new_state": state}, time_fired=12345)
         body = [
             {
-                "measurement": f"fake.{entity_id}",
-                "tags": {"domain": "fake", "entity_id": entity_id},
+                "measurement": test.id,
+                "tags": {"domain": domain, "entity_id": entity_id},
                 "time": 12345,
                 "fields": {"value": 1},
             }
         ]
-        handler_method(event)
-        hass.data[influxdb.DOMAIN].block_till_done()
+        listener.handler(event)
+        hass.data[DOMAIN].block_till_done()
+        print(f"{test} - {write_api.call_count}")
 
-        write_api = get_write_api(mock_client)
-        if entity_id == "ok":
-            assert write_api.call_count == 1
+        if test.should_pass:
+            write_api.assert_called_once()
             assert write_api.call_args == get_mock_call(body)
         else:
             assert not write_api.called
         write_api.reset_mock()
 
+    listener.shutdown()
+
 
 @pytest.mark.parametrize(
     "mock_client, config_ext, get_write_api, get_mock_call",
     [
         (
-            influxdb.DEFAULT_API_VERSION,
+            DEFAULT_API_VERSION,
             BASE_V1_CONFIG,
             _get_write_api_mock_v1,
-            influxdb.DEFAULT_API_VERSION,
+            DEFAULT_API_VERSION,
         ),
+        (API_VERSION_2, BASE_V2_CONFIG, _get_write_api_mock_v2, API_VERSION_2,),
+    ],
+    indirect=["mock_client", "get_mock_call"],
+)
+async def test_event_listener_denylist(
+    hass, mock_client, config_ext, get_write_api, get_mock_call
+):
+    """Test the event listener against a denylist."""
+    config = {"exclude": {"entities": ["fake.denylisted"]}}
+    config.update(config_ext)
+    listener = await _setup(hass, config)
+    write_api = get_write_api(mock_client)
+
+    tests = [
+        FilterTest("fake.ok", True),
+        FilterTest("fake.denylisted", False),
+    ]
+    execute_filter_test(hass, tests, listener, write_api, get_mock_call)
```

Comment:
As mentioned in the description, this PR introduces a breaking change to filtering. This in addition to switching to async means the filter tests all needed some adjustment. I also decided to take the opportunity to simplify them by refactoring a lot of duplicated logic to a helper method.